noah/src/main/resources/crawler/health_map_crawler.py

```diff
@@ -0,0 +1,116 @@
+    dateCounter = datetime.date.today()
+    #get the stop condition date
+    condition = (dateCounter - datetime.timedelta(int(days))).strftime("%m/%d/%Y")
+    # set interval to 1 day
+    interval = datetime.timedelta(1)
+    return dateCounter, condition, interval
+
+def getStartAndEndDate(dateCounter, interval):
+    #get the end date and start date of 1 days interval
+    endDate = dateCounter.strftime("%m/%d/%Y")
+    startDate = (dateCounter - interval).strftime("%m/%d/%Y")
+    return endDate, startDate 
+     
+#get health map response object
+def getHealthMapResponse(url, headers):
+    request = urllib2.Request(url, headers = headers)
+    response = urllib2.urlopen(request)
+    return response
+
+# create the parameters, needs start_date and end_date in "mm/dd/yyyy" format
+def generateHealthMapQuery(start_date, end_date):
+    parameter_map = {"category":["1","2","29"],"diseases":["282"], "heatscore":"1", "partner":"hm"}
+    parameter_map['sdate'] = start_date
+    parameter_map['edate'] = end_date
+    return parameter_map   
+    
+def getEachUrl(alertid, queryParamTo, queryParamFr, pid):
+    query = {}
+    query['trto'] = queryParamTo
+    query['trfr'] = queryParamFr
+    url = "http://www.healthmap.org/ai.php?"+alertid+"&"+urllib.urlencode(query)+"&"+"pid"+pid
+    return url
+    
+def storeJsonResult(completeName, startDate, endDate, headers):
+    with io.open(completeName + ".json", "w", encoding="utf-8") as writeFile:
+            # get response from healthmap
+            url = "http://www.healthmap.org/getAlerts.php?"
+            response = getHealthMapResponse(url + urllib.urlencode(generateHealthMapQuery(startDate, endDate)), headers)
+            #write to file
+            writeFile.write(response.read().decode("utf-8"))
+
+def storeHtmlResult(completeName, headers, newDirName):
+    with io.open(completeName + ".json", "r", encoding="utf-8") as readFile:
+        data = json.load(readFile)
+        for id in data["markers"]:    
+            html = id["html"]
+            (alertid, queryParamTo, queryParamFr, pid) = getQueryParam(html)
+            url = getEachUrl(alertid, queryParamTo, queryParamFr, pid)
+            response = getHealthMapResponse(url, headers)
+            fname = os.path.join(newDirName, alertid)           
+            request = urllib2.Request(url, headers = headers)
+            response = urllib2.urlopen(request)
+            with io.open(fname+".html", "w", encoding="utf-8") as writeFile:
+                writeFile.write(response.read().decode("utf-8"))
+        
+def getQueryParam(html):
+    res = html.find("javascript:b(")
+    start = res + len("javascript:b(")
+    end = html.find(")", start)
+    result = html[start:end]
+    elements = result.split(',')
+    alertid = elements[0]
+    queryParamTo = elements[1].strip('\'')
+    queryParamFr = elements[2].strip('\'')
+    pid = elements[3]
+    return alertid, queryParamTo, queryParamFr, pid
+    
+def directoryCheck(dirName):
```

Comment:
it is not only a `check`. it actually create the dir. should be something like `createDirIfNotExists`.